src/components/ItemMovie/ItemMovie.tsx

```diff
@@ -1,51 +1,98 @@
-import { Content } from '../../constants/common.constants';
-import { IMovie } from '../../types/interfaces';
+import {
+  Content,
+  StarIcon,
+  DefaultValuesCard,
+  ApiErrorMessage,
+} from '../../constants/common.constants';
+import { IMovie, IMovieById } from '../../types/interfaces';
+import notFoundImg from '../../assets/image-not-found.jpg';
+import ModalCard from '../../components/ModalCard/ModalCard';
+import { createPortal } from 'react-dom';
+import { useState } from 'react';
+import Api from '../../Api/Api';
 import classes from './ItemMovie.module.scss';
+import Preloader from '../../components/icons/Preloader/Preloader';
 
 interface IItemMoviesProps {
   key: number;
   movie: IMovie;
-  text: string;
 }
 
-const ItemMovie = ({ text, movie }: IItemMoviesProps): JSX.Element | null => {
-  const { name, poster, year, countries, rating } = movie;
+const ItemMovie = ({ movie }: IItemMoviesProps): JSX.Element | null => {
+  const { id, name, poster, year, countries, rating } = movie;
+  const [showModal, setShowModal] = useState(false);
+  const [movieDetail, setMovieDetail] = useState<IMovieById | null>(null);
+  const [errorMessage, setErrorMessage] = useState<null | string>(null);
+  const [showPreloader, setShowPreloader] = useState(false);
 
-  const foundMovie = (movie: IMovie): boolean => {
-    const { name, year, countries, rating } = movie;
+  const countriesArray = countries ? countries : [];
```

Comment:
u can use default values:
```
const { id, name, poster, year, countries = [], rating } = movie;
```

and add empty line after this destructuring to separate from useState

Reply:
This does not help in this situation, because the server is sending null, not undefined.((